packages/sdk-manager/src/SdkManagerContext.ts

```diff
@@ -0,0 +1,190 @@
+import {
+  EVMAssetTransfer,
+  Environment,
+  EvmFee,
+  Fungible,
+  Transfer
+} from '@buildwithsygma/sygma-sdk-core';
+import { consume, createContext, provide } from '@lit/context';
+import { LitElement, html } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import {
+  WalletManagerContext,
+  WalletManagerController
+} from '@builtwithsygma/sygmaprotocol-wallet-manager';
+import {
+  BaseProvider,
+  Web3Provider,
+  TransactionRequest
+} from '@ethersproject/providers';
+import { UnsignedTransaction } from '@ethersproject/transactions';
+
+export type SdkManagerStatus =
+  | 'idle'
+  | 'initialized'
+  | 'transferCreated'
+  | 'approvalsCompleted'
+  | 'deposited'
+  | 'completed';
+
+export type SdkManagerState = {
+  assetTransfer: EVMAssetTransfer;
+  status: SdkManagerStatus;
+  transfer?: Transfer<Fungible>;
+  fee?: EvmFee;
+  approvalTxs?: UnsignedTransaction[];
+  depositTx?: UnsignedTransaction;
+
+  initialize: (provider: BaseProvider, env?: Environment) => Promise<void>;
+  createTransfer: (
+    provider: BaseProvider,
+    destinationChainId: number,
+    destinationAddress: string,
+    resourceId: string,
+    amount: string
+  ) => Promise<void>;
+
+  performApprovals(provider: Web3Provider): Promise<void>;
```

Comment:
since on the SDK we have `buildApprovals` I think it could make sense to follow that here

Reply:
This function actually sends the transaction to the wallet for signature and broadcast